File: lib/features/technician/domain/tech_team.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'tech_team.freezed.dart';
part 'tech_team.g.dart';

/// Mirrors the backend's `TechTeamResponse`.
@freezed
abstract class TechTeam with _$TechTeam {
  const factory TechTeam({
    required int id,
    required String name,
    required String mono,
    required String oficio,
    required double rating,
    required int reviews,
    required String distance,
    required String priceLabel,
    required double pinTop,
    required double pinLeft,
    required int crew,
    required int projects,
    required int laborCost,
    required int materialsCost,
    required int mobilityCost,
  }) = _TechTeam;

  factory TechTeam.fromJson(Map<String, dynamic> json) => _$TechTeamFromJson(json);
}

File: backend/src/features/technician/domain/tech_team.rs
#[derive(Debug, Clone)]
pub struct TechTeam {
    pub id: i32,
    pub name: String,
    pub mono: String,
    pub oficio: String,
    pub rating: f64,
    pub reviews: i32,
    pub distance: String,
    pub price_label: String,
    pub pin_top: f64,
    pub pin_left: f64,
    pub crew: i32,
    pub projects: i32,
    pub labor_cost: i32,
    pub materials_cost: i32,
    pub mobility_cost: i32,
}
